### PR TITLE
Min-SNR Gamma: correct the fix for SNR weighted loss in v-prediction …

### PR DESCRIPTION
**Pull request by bghira**
…by adding 1 to SNR rather than the resulting loss weights

# What does this PR do?

Fixes issue as mentioned by @parlance-zz [here](https://github.com/huggingface/diffusers/commit/24563ca654f6574dae93aeece8eeef69e39097e5#commitcomment-128670789).

Sorry for all the back and forth on this stuff, I did about 6000 steps of training with the previous fix and all was well. If this works better, awesome!

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
Thanks for iterating!

**Comment by bghira** [comment]
Training two models from scratch to test this one, concurrently. They are using the exact same hyperparameters, with the exception of min-SNR gamma.

With SNR gamma:

image

Without SNR gamma:

image

**Comment by bghira** [comment]
The loss function is very similar but, it has about .1 lower on avg at each step for the gamma model vs the vanilla model. The results themselves seem more substantially-improved than .1 loss would usually indicate.

With gamma:

image

Without gamma:

image

**Comment by bghira** [comment]
the `woman` class:

image

image

**Comment by bghira** [comment]
just for completeness, at 1k steps on both models, the gap is really beginning to widen, where the min-SNR gamma model seriously pulls ahead in terms of representing details and coherence.

image

image

**Comment by feffy380** [comment]
@bghira Doesn't this change reintroduce the division by zero issue when using zero terminal SNR + epsilon prediction?

**Comment by bghira** [comment]
in my opinion, you shouldn't be combining zero terminal SNR with epsilon anyway. i'm sure there's an additional fix that can be applied, but this trainer uses the simplest implementations of the logic from the research paper. using epsilon would be out of scope for that? @sayakpaul thoughts on that?
